eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/common/SwitchYardSettingsGroup.java

```diff
@@ -591,6 +591,9 @@ private void handleRuntimeSelected() {
             final ArtifactVersion version = getRuntimeVersion((IRuntimeComponent) selection.getFirstElement());
             if (version != null) {
                 _runtimeVersionsList.setSelection(new StructuredSelection(version));
+                
+                // to address SWITCHYARD-2968, force the text for Mac OS/X or doesn't get set on selection
+                _runtimeVersionsList.getCombo().setText(version.toString());
```

Comment:
the fix seems strange.
Did you tried to call twice the setSelection ?
there is a SelectionListener which is calling a custom fireChangeEvent, There are 2 ChangeListener registered potentially. Do we still have the issue if we don't fire the change event? Maybe there is somethign doen asynchronuously in these methods?
BTW, one strange thing in ProjectConfigurationWizardPage l.299 , there is a '&' instead of '&&'